SEP-0011.md

```diff
@@ -0,0 +1,154 @@
+- Developer documentation can provide clear instructions to contributors on how to implement *Black* formatting in their development process.
+  Issues with this process are no more difficult (and arguably easier) to resolve than enforcing PEP-8 standards.
+
+Therefore, this SEP proposes to adopt *Black* for the SunPy Project.
+
+#### Concerns and mitigations
+
+##### Foolish consistency
+
+A concern that has been highlighted by some members of the community (including, initially, the author of this SEP, and highlighted in detail in [this blog post](https://luminousmen.com/post/my-unpopular-opinion-about-black-code-formatter) is that adopting *Black* violates the PEP-8 principle that "A Foolish Consistency is the Hobgoblin of Little Minds".
+I.e., that by using a code formatter, the developer thinks less about readability because they think they can let the formatter do this for them.
+While this concern has some validity, this SEP explicitly recognizes that **developers and code reviewers are ultimately responsible for maintaining a readable code base**.
+That is, there are many choices concerning logic and readability of code that *Black* does *not* affect, and those choices are more important for developers to focus on than mechanistic decisions that *Black* makes about whitespace and newlines.
+
+##### Undesirable formatting
+
+On occasion *Black*'s formatting produces undesirable results.
+The code may look very different from expectation due to mathematical or other domain-specific conventions, or some specific formatting that conveys meaning may be lost.
+A simple example of this is a constant identity matrix which would typically be written as
+
+```python
+    matrix = [[1, 0, 0],
+              [0, 1, 0],
+              [0, 0, 1]]
+```
+
+*Black* reformats this as shown below in a way that is very different from the typical representation for a matrix and makes it harder to read the structure
+
+```python
+    matrix = [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
+```
+
+The mitigation here is the exception policy:  In this case the author could surround the matrix with the ``#fmt: off``/``#fmt: on`` escape hatch, as this clearly makes the code more readable.
+While many examples are less clear than the above, and indeed in most cases "readability" is a subjective concept, this
+mechanism allows for such cases to be explicitly excepted where the developer thinks the improved readability warrants it.
+
+##### Migration effort
+
+Enthusiastic champions of the effort will largely do the work, but we recognize and appreciate that additional effort will be required from maintainers that are neutral or against the adoption of *Black*.
+
+#### Backports / release branches
+
+All will be *Black* to ensure that backports are easy to do.
+
+#### New contributors
+
+New contributors may be more discouraged by more steps.
+However, this *replaces* the existing PEP-8 check, so that it is not "another step".
```

Comment:
I think this section really needs rewriting to be more sunpy appropriate, given we already have pre-commit etc.

Reply:
Done